apps/system/js/app_window.js

```diff
@@ -1703,8 +1703,15 @@
 
   AppWindow.prototype.getIconForSplash =
     function aw_getIconForSplash(manifest) {
-      var icons = this.manifest ?
-        ('icons' in this.manifest ? this.manifest.icons : null) : null;
+      var icons;
+      if (this.isPrivate) {
```

Comment:
Hmm, not sure I like faking the icons object with just the single icon - it just feels hackish to me. What do you feel about doing an early return here instead?

```js
if (this.isPrivate) {
  this._splash = '/style/icons/pb_icon.png';
  this.preloadSplash();
  return this._splash;
}
```